From 5067a6a56e86207d85c04fb385f4b9f535a50cd2 Mon Sep 17 00:00:00 2001
From: Sergei Makarov <serg.makarov@samsung.com>
Date: Fri, 5 May 2023 11:37:09 +0300
Subject: [PATCH 18/29] [secndp, pytorch] Use build/installation directory

Reflect changes from SAIT/axdimm_pytorch#61 and use
build directory of PNMLib in SecNDP test and installation
directory of it in PyTorch.

Signed-off-by: Sergei Makarov <serg.makarov@samsung.com>

Related to MCS23-950
---
 application/axdimm_secndp_test/Makefile                |  2 +-
 .../axdimm_secndp_test/secndp.manifest.template        | 10 +++++-----
 application/pytorch/Makefile                           |  2 +-
 application/pytorch/conf.pytorch.manifest.template     |  6 ++----
 4 files changed, 9 insertions(+), 11 deletions(-)

diff --git a/application/axdimm_secndp_test/Makefile b/application/axdimm_secndp_test/Makefile
index 5540a87f..ef6068a3 100644
--- a/application/axdimm_secndp_test/Makefile
+++ b/application/axdimm_secndp_test/Makefile
@@ -17,7 +17,7 @@ secndp.manifest: secndp.manifest.template
 		-Dlog_level=$(GRAMINE_LOG_LEVEL) \
 		-Darch_libdir=$(ARCH_LIBDIR) \
 		-Dentrypoint="${PNM_BUILD_DIR}/test/${AXDIMM_ENTRY}" \
-		-Dpnm_build_dir=${PNM_BUILD_DIR} \
+		-Dpnm_driver=${PNM_BUILD_DIR} \
 		$< > $@
 
 # Make on Ubuntu <= 20.04 doesn't support "Rules with Grouped Targets" (`&:`),
diff --git a/application/axdimm_secndp_test/secndp.manifest.template b/application/axdimm_secndp_test/secndp.manifest.template
index 274f2cb8..2908c146 100644
--- a/application/axdimm_secndp_test/secndp.manifest.template
+++ b/application/axdimm_secndp_test/secndp.manifest.template
@@ -19,9 +19,9 @@ fs.mounts = [
   { uri = "file:/usr", path = "/usr" },
   { uri = "file:/etc", path = "/etc" },
   { uri = "dev:/dev/axdimm_device", path = "/dev/axdimm_device" },
-  { uri = "file:{{ pnm_build_dir }}/test", path = "{{ pnm_build_dir }}/test" },
-  { uri = "file:{{ pnm_build_dir }}/ai", path = "{{ pnm_build_dir }}/ai" },
-  { uri = "file:{{ pnm_build_dir }}/secure", path = "{{ pnm_build_dir }}/secure" },
+  { uri = "file:{{ pnm_driver }}/test", path = "{{ pnm_driver }}/test" },
+  { uri = "file:{{ pnm_driver }}/ai", path = "{{ pnm_driver }}/ai" },
+  { uri = "file:{{ pnm_driver }}/secure", path = "{{ pnm_driver }}/secure" },
   { uri = "file:/sys/class", path = "/sys/class", type="chroot" },
   { uri = "file:/dev/pnm/", path = "/dev/pnm/" },
   { type = "tmpfs", path = "/tmp" }
@@ -47,6 +47,6 @@ sgx.trusted_files = [
   "file:{{ gramine.runtimedir() }}/",
   "file:{{ arch_libdir }}/",
   "file:/usr/{{ arch_libdir }}/",
-  "file:{{ pnm_build_dir }}/ai/",
-  "file:{{ pnm_build_dir }}/secure/",
+  "file:{{ pnm_driver }}/ai/",
+  "file:{{ pnm_driver }}/secure/",
 ]
diff --git a/application/pytorch/Makefile b/application/pytorch/Makefile
index c044c951..d78fa455 100644
--- a/application/pytorch/Makefile
+++ b/application/pytorch/Makefile
@@ -18,7 +18,7 @@ pytorch.manifest: conf.pytorch.manifest.template config_pillow.py
 		-Dlog_level=$(GRAMINE_LOG_LEVEL) \
 		-Darch_libdir=$(ARCH_LIBDIR) \
 		-Dentrypoint=$(realpath $(shell sh -c "command -v python3")) \
-		-Dpnm_build_dir=${PNM_BUILD_DIR} \
+		-Dpnm_driver=${PNM_INSTALL_DIR} \
 		-Ddlrm_path=${DLRM_PATH} \
 		-Dpython_path=${PYTHONPATH} \
 		-Ddlrm_data_path=${DLRM_DATA_PATH} \
diff --git a/application/pytorch/conf.pytorch.manifest.template b/application/pytorch/conf.pytorch.manifest.template
index 13fd7cc2..45004dce 100644
--- a/application/pytorch/conf.pytorch.manifest.template
+++ b/application/pytorch/conf.pytorch.manifest.template
@@ -20,8 +20,7 @@ fs.mounts = [
   { uri = "file:/usr", path = "/usr" },
   { uri = "file:/etc", path = "/etc" },
   #  *** PILLOW MOUNT PLACEHOLDER *** #Will be replaced by config_pillow.py during the make process
-  { uri = "file:{{ pnm_build_dir }}/ai", path = "{{ pnm_build_dir }}/ai"},
-  { uri = "file:{{ pnm_build_dir }}/secure", path = "{{ pnm_build_dir }}/secure"},
+  { uri = "file:{{ pnm_driver }}/lib", path = "{{ pnm_driver }}/lib"},
   { uri = "file:{{ dlrm_data_path }}", path = "/input" },
   { type = "tmpfs", path = "/tmp" },
   { type = "shm" },
@@ -47,8 +46,7 @@ sgx.trusted_files = [
   "file:{{ python.stdlib }}/",
   "file:{{ python.distlib }}/",
   "file:{{ python.get_path('stdlib', vars={'installed_base': '/usr/local'}) }}/",
-  "file:{{ pnm_build_dir }}/ai/",
-  "file:{{ pnm_build_dir }}/secure/"
+  "file:{{ pnm_driver }}/lib/",
 ]
 
 sgx.allowed_files = [
-- 
2.34.1

